internal/util/fetch/fetch.go

```diff
@@ -199,9 +199,14 @@ func ClonerUsingGitExec(ctx context.Context, repoSpec *git.RepoSpec) error {
 		return errors.E(op, errors.Internal, fmt.Errorf("error copying package: %w", err))
 	}
 
+	// Verify that the requested path exists in the repo.
+	pkgPath := filepath.Join(dir, repoSpec.Path)
+	if _, err = os.Stat(pkgPath); os.IsNotExist(err) {
+		return errors.E(op, errors.Internal, err, fmt.Errorf("path %q does not exist in repo %q", repoSpec.Path, repoSpec.OrgRepo))
```

Comment:
@etefera Can you share the final CLI output ? (may be update in the PR description).

Reply:
Updated PR description with the following:
```
$ kpt pkg get git@github.com:GoogleContainertools/kpt.git/bibbidi-bobbidi-boo
Package "bibbidi-bobbidi-boo":
Fetching git@github.com:GoogleContainertools/kpt@main
From github.com:GoogleContainertools/kpt
 * branch            main       -> FETCH_HEAD
Error: path "/bibbidi-bobbidi-boo" does not exist in repo "git@github.com:GoogleContainertools/kpt" ```